rwengine/src/ai/CharacterController.cpp

```diff
@@ -677,8 +677,8 @@ bool Activities::UseItem::update(CharacterObject *character,
     const auto &weapon = world->data->weaponData.at(itemslot);
```

Comment:
What if there is no weapon at itemslot?

Reply:
According to standard, `at` should throw exception. It's connected with question first post. Ref cannot be empty - so if we use this option, then modified and invalid assets could crash game. (ofc pointers also require checks)

Question if is it a problem?

Reply:
I guess not now.
Later on, malicious scm or a scm fuzzer can crash the game.